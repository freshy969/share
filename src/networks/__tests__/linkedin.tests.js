import Linkedin from '../linkedin';

describe('Share > Linkedin', () => {
  test('url', () => {
    const linkedin = Linkedin.create({});
    Object.defineProperty(linkedin, 'entityLink', {value: jest.fn(), writable: true});
    linkedin.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauties-of-gullfoss');

    expect(
      linkedin.url({
        type: 'post',
        id: 60,
        title: 'The Beauties of Gullfoss',
        summary:
          'Gullfoss is a waterfall located in the canyon of the Hvítá river in southwest Iceland.',
      }),
    ).toMatchSnapshot();
  });
});
