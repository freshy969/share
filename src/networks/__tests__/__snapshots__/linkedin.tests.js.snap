// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share > Linkedin requestCount fails 1`] = `
Array [
  Array [
    "https://www.linkedin.com/countserv/count/share?url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss",
  ],
  Array [
    "https://www.linkedin.com/countserv/count/share?url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss",
  ],
]
`;

exports[`Share > Linkedin requestCount success 1`] = `
Array [
  Array [
    Object {
      "id": 60,
      "type": "post",
    },
  ],
]
`;

exports[`Share > Linkedin requestCount success 2`] = `
Array [
  Object {
    "isThrow": false,
    "value": "https://demo.frontity.com/the-beauties-of-gullfoss",
  },
]
`;

exports[`Share > Linkedin requestCount success 3`] = `
Array [
  Array [
    "https://www.linkedin.com/countserv/count/share?url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss",
  ],
]
`;

exports[`Share > Linkedin requestCount success 4`] = `
Array [
  Object {
    "isThrow": false,
    "value": Promise {},
  },
]
`;

exports[`Share > Linkedin url 1`] = `"https://www.linkedin.com/shareArticle?url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss&title=The%20Beauties%20of%20Gullfoss&summary=Gullfoss%20is%20a%20waterfall%20located%20in%20the%20canyon%20of%20the%20Hv%C3%ADt%C3%A1%20river%20in%20southwest%20Iceland."`;
